src/go/wsl-helper/cmd/k3s_kubeconfig.go

```diff
@@ -78,49 +74,54 @@ var k3sKubeconfigCmd = &cobra.Command{
 		case configFile = <-ch:
 			break
 		}
-
-		var config kubeConfig
-		defer configFile.Close()
-		err = yaml.NewDecoder(configFile).Decode(&config)
+		config, err := updateClusterIP(configFile, false)
 		if err != nil {
 			return err
 		}
-
-		if rdNetworking {
-			// vm-switch in rdNetworking binds to localhost:Port by default.
-			// Since k3s.yaml comes with servers preset at 127.0.0.1, there
-			// is nothing for us to do here, just write the config and return.
-			return yaml.NewEncoder(os.Stdout).Encode(config)
-		}
-		ip, err := getClusterIP()
+		// Emit the result
+		err = yaml.NewEncoder(os.Stdout).Encode(config)
 		if err != nil {
 			return err
 		}
-		// Fix up any clusters at 127.0.0.1, using the IP address we found.
-		for clusterIdx, cluster := range config.Clusters {
-			server, err := url.Parse(cluster.Cluster.Server)
-			if err != nil {
-				// Ignore any clusters with invalid servers
-				continue
-			}
-			if server.Hostname() != "127.0.0.1" {
-				continue
-			}
+		return nil
+	},
+}
+
+func updateClusterIP(configFile *os.File, rdNetworking bool) (kubeConfig, error) {
+	var config kubeConfig
+	defer configFile.Close()
+	err := yaml.NewDecoder(configFile).Decode(&config)
+	if err != nil {
+		return config, err
+	}
+
+	ip, err := getClusterIP()
```

Comment:
We should delay getting this (or use a [`sync.OnceValue`](https://pkg.go.dev/sync@go1.21.3#OnceValue) or something), so that if we fail to get eth0 in an alternative distro we don't prematurely fall over if we don't need it.